gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py

```diff
@@ -0,0 +1,109 @@
+
+DEFAULT_WINDOW_SIZE = 2500
+
+
+class MergeWithNonVariantsStrategy(variant_merge_strategy.VariantMergeStrategy):
+  """A merging strategy that can handle non-variant records.
+  Variants will be merged across files by using reference_name and start
+  position as key. INFO fields should be moved to class if they match
+  `info_keys_to_move_to_calls_regex`. Otherwise, one will be chosen as
+  representatve (in no particular order) among the merged variants.
+  Filters will be merged across all variants matching the key and the
+  highest quality score will be chosen as representative for the merged
+  variants. The filters and quality fields can be optionally copied to
+  their associated calls using `copy_quality_to_calls` and
+  `copy_filter_to_calls` options.
+  Note: if a field is set to be moved from INFO to calls, then it must
+  not already exist in calls (i.e. specified by FORMAT in the VCF
+  header).
+  """
+
+  def __init__(self, info_keys_to_move_to_calls_regex, copy_quality_to_calls,
+               copy_filter_to_calls, window_size=DEFAULT_WINDOW_SIZE):
+    """Initializes the strategy.
+    Args:
+      info_keys_to_move_to_calls_regex (str): A regular expression
+        specifying info fields that should be moved to calls.
+      copy_quality_to_calls (bool): Whether to copy the quality field to
+        the associated calls in each record.
+      copy_filter_to_calls (bool): Whether to copy filter field to the
+        associated calls in each record.
+      window_size (int): Size of windows that variants will be grouped
+        in based on the start position of the variant
+    """
+    self._move_to_calls_strategy = move_to_calls_strategy.MoveToCallsStrategy(
+        info_keys_to_move_to_calls_regex=info_keys_to_move_to_calls_regex,
+        copy_quality_to_calls=copy_filter_to_calls,
+        copy_filter_to_calls=copy_filter_to_calls)
+    self._window_size = window_size
+
+  def _union(self, list1, list2):
+    return sorted(set(list1 + list2))
+
+  def _merge(self, v1, v2):
+    if v1.alternate_bases == v2.alternate_bases:
+      merged_variant = Variant(reference_name=v1.reference_name,
+                               start=v1.start,
+                               end=v1.end,
+                               reference_bases=v1.reference_bases,
+                               alternate_bases=v1.alternate_bases,
+                               names=self._union(v1.names, v2.names),
+                               filters=self._union(v1.filters, v2.filters),
+                               quality=max(v1.quality, v2.quality),
+                               calls=sorted(v1.calls + v2.calls))
+      self._move_to_calls_strategy.move_data_to_merged(v1, merged_variant)
+      self._move_to_calls_strategy.move_data_to_merged(v2, merged_variant)
+      yield merged_variant
+    else:
+      yield v1
+      yield v2
+
+  def get_merged_variants(self, variants):
+    tree = IntervalTree()
```

Comment:
Please add a TODO for handling non-variant regions. It's not obvious that this is WIP :)

Reply:
Done.